### PR TITLE
[FEAT] Pet UI

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR adds the UI for the Faction Pets.

NOTE: I also have the claim emblems cutoff check in this PR

Fixes #issue

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- Run against the BE

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [ ] I have read the contributing guidelines and agree to the T&Cs
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/world/scenes/FactionHouseScene.ts [hunk 1]
```diff
@@ -3,12 +3,63 @@ import { interactableModalManager } from "../ui/InteractableModals";
...
+      clearInterval(this.fetchInterval);
+    }
+  }
+
+  getPetState() {
+    const week = getFactionWeek({ date: new Date() });
+    const beginningOfWeek = new Date(week).getTime();
+    const firstWeek = "2024-07-08";
+
+    if (!this.collectivePet || week === firstWeek) return "pet_hungry";
```
Comment (brian-man) [inline src/features/world/scenes/FactionHouseScene.ts:51]: The pet is "hungry" for the entire first week no matter how much it's fed?

File: src/features/world/scenes/FactionHouseScene.ts [hunk 2]
```diff
@@ -3,12 +3,63 @@ import { interactableModalManager } from "../ui/InteractableModals";
...
   preload() {
     super.preload();
 
     this.load.image("basic_chest", "world/basic_chest.png");
+    this.makeFetchRequest();
+  }
+
+  create() {
+    super.create();
+    this.fetchInterval = setInterval(() => this.makeFetchRequest(), 10 * 1000);
```
Comment (brian-man) [inline src/features/world/scenes/FactionHouseScene.ts:24]: Should this be using `REFRESH_INTERVAL`?

**Review by c-py — APPROVED** [review]
LGTM! Tested on two browsers and had the pet XP correctly update

File: src/features/world/scenes/FactionHouseScene.ts (hunk 1 above)
Comment (c-py) [reply src/features/world/scenes/FactionHouseScene.ts:57]: Yep, the pet can be overfed

**Pull request merged**